pinot-spi/src/main/java/org/apache/pinot/spi/stream/MessageBatch.java

```diff
@@ -104,6 +104,15 @@ default StreamPartitionMsgOffset getNextStreamPartitionMsgOffsetAtIndex(int inde
     return new LongMsgOffset(getNextStreamMessageOffsetAtIndex(index));
   }
 
+  /**
+   * Returns the offset of this message.
+   * @param index
+   * @return
+   */
+  default StreamPartitionMsgOffset getStreamPartitionMsgOffsetAtIndex(int index) {
+    return new LongMsgOffset(getMessageOffsetAtIndex(index));
```

Comment:
@Jackie-Jiang : at present there's no usage of `MessageBatch#getMessageOffsetAtIndex` anywhere in the Pinot code. At present we use `getNextStreamPartitionMsgOffsetAtIndex` which uses the deprecated method `getNextStreamMessageOffsetAtIndex`.

If a user has a `MessageBatch` implementation that sits outside of this repo, and they haven't implemented this method, then they will run into errors.

However, all of the `MessageBatch` implementations in this repo implement this method and hence should continue to work as expected.

Given that, let us know if this is still okay to go.

Reply:
That is true, and I'm trying to make a PR to clean up these APIs. I'd also suggest not adding new APIs for now.
@tibrewalpratik17 Is this change urgent? If it is not urgent, I'd like to first clean up this interface, then you may use the new APIs

Reply:
> @tibrewalpratik17 Is this change urgent? If it is not urgent, I'd like to first clean up this interface, then you may use the new APIs

Sure can wait until your changes

Reply:
@Jackie-Jiang updated based on your recent refactoring. Please review!